src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,196 @@
+
+            if (size == 0) {
+                first = newNode;
+                last = newNode;
+            } else {
+                last.next = newNode;
+                last = newNode;
+            }
+
+            size++;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        Node n = first;
+        for (int i = 0; i < index; i++) {
+            n = n.next;
+        }
+        return (T) n.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        Node n = first;
+        for (int i = 0; i < index; i++) {
+            n = n.next;
+        }
+
+        T oldValue = (T) n.item;
+        n.item = value;
+
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        T delItem;
+
+        if (index == 0) {
+            delItem = (T) first.item;
+            first = first.next;
+            if (first != null) {
+                first.prev = null;
+            } else {
+                last = null;
+            }
+        } else if (index == size - 1) {
+            delItem = (T) last.item;
+            last = last.prev;
+            if (last != null) {
+                last.next = null;
+            } else {
+                first = null;
+            }
+        } else {
+            Node n = first;
+            for (int i = 0; i < index - 1; i++) {
+                n = n.next;
+            }
+            Node delNode = n.next;
+            delItem = (T) delNode.item;
+
+            n.next = delNode.next;
+            if (delNode.next != null) {
+                delNode.next.prev = n;
+            }
+        }
+
+        size--;
+        return delItem;
     }
 
     @Override
     public boolean remove(T object) {
+        Node node = first;
+
+        while (node != null) {
+            if (node.item != null && node.item.equals(object) || node.item == object) {
+                if (node.prev != null) {
+                    node.prev.next = node.next;
+                } else {
+                    first = node.next;
+                }
+
+                if (node.next != null) {
+                    node.next.prev = node.prev;
+                } else {
+                    last = node.prev;
+                }
+
+                size--;
+                return true;
+            }
+            node = node.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<E> {
```

Comment:
The `Node` class should be generic with the same type parameter `T` as the `MyLinkedList` class. Change `Node<E>` to `Node<T>` to ensure type consistency.